packages/terra-collapse/tests/jest/Collapse.test.jsx

```diff
@@ -0,0 +1,134 @@
+    expect(collapse).toMatchSnapshot();
+  });
+
+  // Event Tests
+  it('should render an open collapse when clicked', () => {
+    const collapse = mount(<Collapse closedButtonText="Show">Test</Collapse>);
+    collapse.find('button').simulate('click');
+    expect(collapse).toMatchSnapshot();
+  });
+
+  // Prop Tests
+  it('should set buttonAttrs prop correctly', () => {
+    const collapse = shallow(<Collapse buttonAttrs={{ variant: 'link' }} closedButtonText="Show">Test</Collapse>);
+    expect(collapse.unrendered.props.buttonAttrs).toEqual({ variant: 'link' });
+  });
+
+  it('should set children prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show">Test</Collapse>);
+    expect(collapse.unrendered.props.children).toEqual('Test');
+  });
+
+  it('should set closedButtonText prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show">Test</Collapse>);
+    expect(collapse.unrendered.props.closedButtonText).toEqual('Show');
+  });
+
+  it('should set icon prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show">Test</Collapse>);
+    expect(collapse).toMatchSnapshot();
+  });
+
+  it('should set icon prop correctly with custom icon', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" icon={<IconCaretRight />}>Test</Collapse>);
+    expect(collapse).toMatchSnapshot();
+  });
+
+  it('should set isAnimated prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" isAnimated>Test</Collapse>);
+    expect(collapse.unrendered.props.isAnimated).toEqual(true);
+  });
+
+  it('should set isInitiallyOpen prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" isInitiallyOpen>Test</Collapse>);
+    expect(collapse.unrendered.props.isInitiallyOpen).toEqual(true);
+  });
+
+  it('should set onOpen prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" onOpen={() => alert('Opened')}>Test</Collapse>);
+    expect(collapse.unrendered.props.onOpen.toString()).toEqual('function onOpen() {return alert(\'Opened\');}');
+  });
+
+  it('should set onClose prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" onClose={() => alert('Closed')}>Test</Collapse>);
+    expect(collapse.unrendered.props.onClose.toString()).toEqual('function onClose() {return alert(\'Closed\');}');
+  });
+
+  it('should set openedButtonText prop correctly', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show" openedButtonText="Hide">Test</Collapse>);
+    expect(collapse.unrendered.props.openedButtonText).toEqual('Hide');
+  });
+
+  // Structure Tests
+  it('should have the class terra-Collapse', () => {
+    const collapse = shallow(<Collapse closedButtonText="Show">Test</Collapse>);
+    expect(collapse.prop('className')).toContain('terra-Collapse');
```

Comment:
For these id and className tests, doesn't the jest snapshot check these already when you are doing snapshot comparisons? It feels like your other tests are already testing for this.